Infra - Trigger CI when pushing to a release branch or tag

kbendick opened the pull request:
We'd like to run CI to completion when we push to a release branch.

Release branches are of the form `0.x.y` right now, so matching on `0.**`

For release tags, they are normally `apache-iceberg-**`, so I've updated the CI suites to run on push to a tag of that format.

cc @rdblue @snazy @nastra

rdblue commented:
Sorry, I think I was wrong. Branches are named after the release version, like `0.13.x`. The release tags are named `apache-iceberg-**`, but we already run on tags. So I think it should be `0.**`?

kbendick commented:
> Sorry, I think I was wrong. Branches are named after the release version, like `0.13.x`. The release tags are named `apache-iceberg-**`, but we already run on tags. So I think it should be `0.**`?

No worries. I updated it to be branches of `0.**` (not sure how to require that it be `0.x.y` formatted and realistically before this we were running CI on push to any branch).

I also updated the tags to be `apache-iceberg-**` since that is the format for release tags.

snazy reviewed (approved):
LGTM
